numba/np/arraymath.py

```diff
@@ -4159,6 +4159,27 @@ def np_select_arr_impl(condlist, choicelist, default=0):
 
     return np_select_arr_impl
 
+
+@overload(np.asarray_chkfinite)
+def np_asarray_chkfinite(a, dtype=None, order='C'):
+
+    msg = "The argument to np.asarray_chkfinite must be array-like"
+    if not isinstance(a, (types.Array, types.Sequence, types.Tuple)):
+        raise TypingError(msg)
+
+    if is_nonelike(dtype):
+        dt = a.dtype
+    else:
+        dt = dtype.dtype
+
+    def impl(a, dtype=None, order='C'):
+        a = np.asarray(a, dtype=dt)
+        if not np.all(np.isfinite(a)):
```

Comment:
The Numpy docs state that the exception is raised on Nan and infinity, so it's probably best to check for NaN here  too.

https://numpy.org/doc/stable/reference/generated/numpy.asarray_chkfinite.html

Reply:
Please also see the comment about implementation above.

Reply:
I have a doubt here according to [this](https://numpy.org/doc/stable/reference/generated/numpy.isfinite.html) isfinite checks for both infinity and NaN. Should I check for NaNs along with this or use [np.isnan](https://numpy.org/doc/stable/reference/generated/numpy.isnan.html) and [np.isinf](https://numpy.org/doc/stable/reference/generated/numpy.isinf.html) ?

Reply:
ah yes, my bad, please do use `np.isinf`.